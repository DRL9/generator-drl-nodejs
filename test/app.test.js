const path = require('path');
const helpers = require('yeoman-test');
const fs = require('fs');

describe('app', () => {
    const generatorPath = path.join(__dirname, '../generators/app');
    test('run sucess', () => expect(helpers.run(generatorPath)).resolves.toBeTruthy());
    test('creates expected files', () => {
        return helpers
            .run(generatorPath)
            .then(() =>
                Promise.all(
                    [
                        '.eslintignore',
                        '.eslintrc',
                        '.gitignore',
                        'jsconfig.json',
                        '.prettierrc',
                        'index.js',
                        '.huskyrc.js',
                        'jest.config.js',
                        'src',
                        'test',
                    ].map((file) => expect(fs.promises.stat(file)).resolves.toBeTruthy())
                )
            );
    });
    test('projectName is rendered', () => {
        return helpers
            .run(generatorPath)
            .withPrompts({
                projectName: 'myProjectName',
            })
            .then(() =>
                Promise.all([
                    expect(fs.promises.readFile('package.json', 'utf8').then(JSON.parse)).resolves.toMatchObject({
                        name: 'myProjectName',
                        description: 'myProjectName',
                    }),
                    expect(fs.promises.readFile('readme.md', 'utf8')).resolves.toMatch(/# myProjectName/),
                ])
            );
    });
    test('has git', () => {
        return helpers
            .run(generatorPath)
            .then(() => expect(fs.promises.stat('.git').then((s) => s.isDirectory())).resolves.toBeTruthy());
    });
});
