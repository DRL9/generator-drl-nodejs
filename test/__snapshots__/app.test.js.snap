// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init Git deps is right 1`] = `
Array [
  "@types/jest",
  "eslint",
  "eslint-config-prettier",
  "eslint-plugin-jest",
  "eslint-plugin-prettier",
  "jest",
  "prettier",
  "husky",
]
`;

exports[`select framework Koa deps is right 1`] = `
Array [
  "@types/jest",
  "eslint",
  "eslint-config-prettier",
  "eslint-plugin-jest",
  "eslint-plugin-prettier",
  "jest",
  "prettier",
  "nodemon",
  "husky",
  "@koa/cors",
  "@koa/router",
  "koa",
  "koa-bodyparser",
  "winston",
  "winston-daily-rotate-file",
  "chalk",
]
`;

exports[`select framework None deps is right 1`] = `
Array [
  "@types/jest",
  "eslint",
  "eslint-config-prettier",
  "eslint-plugin-jest",
  "eslint-plugin-prettier",
  "jest",
  "prettier",
  "husky",
]
`;
